apps/dashboard/config/configuration_singleton.rb

```diff
@@ -250,8 +250,54 @@ def native_vnc_login_host
     ENV['OOD_NATIVE_VNC_LOGIN_HOST']
   end
 
+  # Set the global configuration file
+  def config_file
+    Pathname.new(ENV['OOD_CONFIG_FILE'] || "/etc/ood/config/config.yml")
```

Comment:
Might want to get @treydock feedback on the location of this new config file. I know I suggested "/etc/ood/config/config.yml" but should it be "/etc/ood/config/apps/config.yml" instead or named something else? @treydock this is where we would place configuration for "pinned apps" and then start adding support for other things we currently use env vars for - like the Help menu links and branding options.

Reply:
I would not recommend `/etc/ood/config/apps` because that could interfere with our currently simply method of syncing app configs from git to that location.  I think the best solution is `/etc/ood/config/config.yml` but if this file will be a general ondemand config maybe change filename from `config.yml` to something like `ondemand-apps.yml` or something a little more descriptive.  I can make the `/etc/ood/config/apps` work but it's mostly a behavior with how Puppet handles recursive configs so worry other configuration management tools might not handle it as well.

Reply:
For right now we intend to only use this for ondemand app config. If in the future this config is also read by nginx_stage and even Apache lua code would that be problematic in terms of naming it ondemand-apps.yml? For example, if branding information is in this config, we could use this to standardize the look and feel of all error messages, whether or not they are produced by Lua script, by the PUN NGINX, or by an individual Rails app.

Reply:
In that case I think maybe just `ondemand.yaml`.  It just looks funny to have the path be `config/config.yaml`, where dirname is same as basename without extension.  I can tell you from experience with dealing with projects that use YAML that stuffing too much in 1 YAML file can become hard to maintain.  I would recommend supporting a directory of YAML so like `/etc/ood/config/ondemand.d/*.yml` or something so that people can split their configs into smaller files for easier maintainability.

Reply:
Yea I would +1 a `/etc/ood/config/ondemand.d/*.yml` scheme.

Reply:
Ideally if the config file is `/etc/ood/config/ondemand.yml` the dot D directory would be same but replace `yml` with `d` to keep things consistent.